### PR TITLE
parse(IPAddr, str): use the generic parse function for IP parsing

### PR DESCRIPTION
**Pull request by StefanKarpinski**
Likewise `parse(IPv4, str)` and `parse(IPv6, str)`.

### CODE REVIEW

**Comment by tkelman** [comment]
could use a deprecation

File: base/docs/helpdb/Base.jl [hunk 1]
```diff
@@ -4049,13 +4049,6 @@ itself). For matrices, returns an identity matrix of the appropriate size and ty
 one
 
 """
-    parseip(addr)
-
-Parse a string specifying an IPv4 or IPv6 ip address.
```
Comment (tkelman) [inline base/docs/helpdb/Base.jl]: worth tweaking this to be a method-specific docstring for parse?

also needs to be removed from the rst

**Comment by tkelman** [comment]
also in contrib/BBEditTextWrangler-julia.plist

**Comment by JeffBezanson** [comment]
+1

**Pull request merged**